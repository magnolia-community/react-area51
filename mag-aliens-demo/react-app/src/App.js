import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import axios from "axios";

import { Area51Context } from "magnolia-react-area51";
import { Page } from "magnolia-react-area51";

import Footer from "./app/component/Footer";
import Navigation from "./app/component/Navigation";

import { dlog, inPageEditor } from "./app/AppHelpers";

import ENVIRONMENT from "./environments/environment";
import COMPONENTS from "./app/mapping";

import PAGE_MODEL from "./sampleData/page-model";
import TEMPLATE_DEFINITIONS from "./sampleData/template-definitions";

class App extends Component {
  /**
   * On init
   */
  constructor(props) {
    super(props);

    dlog("***");
    dlog("App Constructor.");

    this.USE_SAMPLE_DATA = false;

    if (this.props.useSampleData) {
      this.USE_SAMPLE_DATA = true;
    }

    this.loadPageContent = this.loadPageContent.bind(this);

    // Initialize state object.
    this.state = {
      init: false,
      inPageEditor: false,
      rootCmsPath: ENVIRONMENT.rootCmsPath,
      serverPath: ENVIRONMENT.serverPath,
      componentMap: COMPONENTS,
      content: {},
      templateDefinitions: {}
    };
  }

  /**
   * Once the component and all his children are rendered
   */
  componentDidMount() {
    dlog("App. componentDidMount.");

    if (this.USE_SAMPLE_DATA) {
      this.props.history.push("/mars/phobos");
      this.useSampleData();
    } else {
      if (window.singlePageConfig != null) {
        //App is running in Magnolia page.

        this.setState({
          init: true,
          inPageEditor: inPageEditor(),
          content: window.singlePageConfig.content,
          templateDefinitions: window.singlePageConfig.templateDefinitions
        });
      } else {
        var h = this.props.history.location.pathname;
        //this.loadPageContent('/')
        this.loadPageContent(h);
      }
    }
  }

  loadPageContent(relativePath) {
    dlog("loadPageContent:" + relativePath);
    if (this.USE_SAMPLE_DATA) {
      return;
    }

    relativePath = this.removeExtension(relativePath);

    // if (relativePath.lastIndexOf(".") > 0) {
    //   relativePath = relativePath.substr(0, relativePath.lastIndexOf("."));
    // }

    this.setState({ init: false });

    var fullURL =
      ENVIRONMENT.restUrlBase + ENVIRONMENT.rootCmsPath + relativePath;
    dlog("Request content from: " + fullURL);

    //Loads the single page config
    axios
      .get(fullURL)
      .then(response => {
        dlog("***");
        dlog("Content: got magnolia page content.");

        this.setState({
          init: true,
          inPageEditor: inPageEditor(),
          content: response.data,

          /* TODO If we every want edit (or debug, or test)
          outside the context of a magnolia page template - 
        then we Also need an endpoint to get the template definitions. */
          templateDefinitions: null
        });
      })
      .catch(error => {
        dlog(error);
      });
  }

  useSampleData() {
    dlog("***");
    dlog("Content: Use Sample Data.");

    this.setState({
      init: true,
      inPageEditor: inPageEditor(),
      content: PAGE_MODEL,
      templateDefinitions: TEMPLATE_DEFINITIONS
    });
  }

  getEditorPath(relativePath) {
    // remove serverPath
    relativePath = relativePath.substr(ENVIRONMENT.serverPath.length);

    // remove rootCmsPath
    relativePath = relativePath.substr(ENVIRONMENT.rootCmsPath.length);
    return relativePath;
  }

  removeExtension(path) {
    if (path.indexOf(".") > -1) {
      path = path.substr(0, path.lastIndexOf("."));
    }
    return path;
  }

  componentWillMount() {
    // Use ReactRouter to handle route events when the browser URL changes.
    this.unlisten = this.props.history.listen((location, action) => {
      var relativePath = location.pathname;

      if (this.state.inPageEditor) {
        relativePath = this.getEditorPath(relativePath);
      }

      relativePath = this.removeExtension(relativePath);

      dlog("***");
      dlog("Route Change. RelativePath: " + relativePath);

      this.loadPageContent(relativePath);
    });
  }
  componentWillUnmount() {
    //this.unlisten();
  }

  render() {
    if (this.state.init) {
      dlog("***");
      dlog("App Render.");

      return (
        <div>
          <Navigation />
          <Area51Context.Provider value={this.state}>
            <Page />
            {/*<PageHome/>*/}

            <Footer />
          </Area51Context.Provider>
        </div>
      );
    } else {
      return (
        <div>
          <Navigation />
          <p>Loading content from CMS...</p>
        </div>
      );
    }
  }
}
export default withRouter(App);
