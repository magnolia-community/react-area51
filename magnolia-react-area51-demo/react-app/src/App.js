import React, { Component } from 'react';
import {withRouter } from 'react-router-dom'
import axios from 'axios';

import {Area51Context} from 'magnolia-react-area51';
import {Page} from 'magnolia-react-area51';

import Footer from './app/component/Footer';
import Navigation from './app/component/Navigation';

import {dlog} from './app/AppHelpers';

import ENVIRONMENT from "./environments/environment"
import COMPONENTS from './app/mapping'

import PAGE_MODEL from './sampleData/page-model';
import TEMPLATE_DEFINITIONS from './sampleData/template-definitions';




class App extends Component {

	/**
	 * On init
	 */
	constructor(props) {
		super(props);

		dlog('***');
		dlog("App Constructor.")

		this.USE_SAMPLE_DATA = false;

		this.loadPageContent = this.loadPageContent.bind(this);

		// Initialize state object.
		this.state = {
			init: false,
			inPageEditor: false,
			rootCmsPath: ENVIRONMENT.rootCmsPath,
			componentMap:COMPONENTS,
			content: {},
			templateDefinitions: {}
		};
	}

	/**
	 * Once the component and all his children are rendered
	 */
	componentDidMount() {
		dlog("App. componentDidMount.");

		if (this.USE_SAMPLE_DATA){
			this.useSampleData();
		}else{

			if (window.singlePageConfig != null) {
				//App is running in Magnolia page.

				this.setState( {
					init: true,
					inPageEditor: this.inPageEditor(),
					rootCmsPath: ENVIRONMENT.rootCmsPath,
					componentMap:COMPONENTS,
					content: window.singlePageConfig.content,
					templateDefinitions: window.singlePageConfig.templateDefinitions
				} );
				
			} else {
				var h = this.props.history.location.pathname;
				//this.loadPageContent('/')
				this.loadPageContent(h);
			}
		}

	}

	loadPageContent(relativePath){

		dlog("loadPageContent:" + relativePath);
		if (relativePath.lastIndexOf('.')>0){
			relativePath = relativePath.substr(0, relativePath.lastIndexOf('.'));
		}

		this.setState ( {init:false});

		if (this.USE_SAMPLE_DATA){
			this.useSampleData();
		}else{

			//Loads the single page config
			axios
				.get(ENVIRONMENT.restUrlBase + ENVIRONMENT.rootCmsPath + relativePath)
				.then(response => {
					dlog('***');
					dlog("Content: got magnolia page content.")

					this.setState( {
						init: true,
						inPageEditor: this.inPageEditor(),
						rootCmsPath: ENVIRONMENT.rootCmsPath,
						componentMap: COMPONENTS,
						content: response.data,

					/* TODO If we every want edit (or debug, or test)
						outside the context of a magnolia page template - 
					then we Also need an endpoint to get the template definitions. */
						templateDefinitions: null
					} );


				}).catch(error => {
				    dlog(error);
				});
			}
	}

	useSampleData(){
		dlog('***');
		dlog("Content: Use Sample Data.");

		this.setState( {
			init: true,
			inPageEditor: this.inPageEditor(),
			rootCmsPath: ENVIRONMENT.rootCmsPath,
			componentMap:COMPONENTS,
			content: PAGE_MODEL,
			templateDefinitions: TEMPLATE_DEFINITIONS
		} );
	}

	inPageEditor(){
		if (window.inPageEditor === true){
			return true;
		} else{
			return false;
		}
	}


	componentWillMount() {

		// Use ReactRouter to handle route events when the browser URL changes.
    this.unlisten = this.props.history.listen((location, action) => {
			
			var relativePath = 	location.pathname;
			if (relativePath.indexOf('.')>-1){
				relativePath = relativePath.substr(0, relativePath.lastIndexOf('.'));
			}

			dlog('***');
			dlog("Route Change. RelativePath: " + relativePath);

			this.loadPageContent(relativePath)
    });
  }
  componentWillUnmount() {
      //this.unlisten();
  }


	render() {

		if ( this.state.init ) {
			dlog('***');
			dlog("App Render.")
			
			return (
				<div>
				<Area51Context.Provider value={this.state}>

					<Navigation/>

					<Page/>
					{/*<PageHome/>*/}

					<Footer/>

				</Area51Context.Provider>
				</div>
			);
    } else {
      return null;
    }
	}

}
export default withRouter(App);

