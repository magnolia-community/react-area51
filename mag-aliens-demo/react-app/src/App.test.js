import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
});

/**
 * Simple Snapshot tet for the entire app - based on the contents of the sampleData directory.
 */
it("App renders correctly with sampleData", () => {
  const tree = renderer.create(<App useSampleData={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});
