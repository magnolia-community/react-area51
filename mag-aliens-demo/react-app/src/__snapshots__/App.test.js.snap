// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly with sampleData 1`] = `
<div>
  <div>
    <div
      className="content-background"
    >
      <div
        className="container"
      >
        <h1
          className="bd-title"
        >
          Phobos
        </h1>
        <div
          className="component-name"
        >
          Page Standard
        </div>
        <div>
          <div />
        </div>
        <div
          className="row mgn-train-section"
        >
          <div
            className="col-xs-12 col-md-6"
          >
            <div>
              
              <div
                className="slideshow"
              >
                <h4>
                  Best Aliens (Phobos)
                </h4>
                <div
                  className="component-name"
                >
                  Slideshow
                </div>
                <div>
                  
                  <div
                    className="card slide"
                    style={
                      Object {
                        "width": "18rem",
                      }
                    }
                  >
                    <div
                      className="card-body"
                    >
                      <h5
                        className="card-title"
                      >
                        Porg
                      </h5>
                      <p
                        className="card-text"
                      >
                        Much, much stupider then as portrayed in the Star Wars franchise.
                      </p>
                    </div>
                  </div>
                  <div
                    className="card slide"
                    style={
                      Object {
                        "width": "18rem",
                      }
                    }
                  >
                    <div
                      className="card-body"
                    >
                      <h5
                        className="card-title"
                      >
                        Phorly
                      </h5>
                      <p
                        className="card-text"
                      >
                        These microbial beings have evolved spiral formed bodies to efficiently extract nutrients from rock.
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-xs-12 col-md-6"
          >
            <div>
              
              <div>
                <h5
                  className="light"
                >
                  It's a moon of Mars.
                </h5>
                <div />
                <div>
                  <p
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": undefined,
                      }
                    }
                  />
                </div>
                <div
                  className="component-name"
                >
                  TextImage
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="row mgn-train-section"
        >
          <div
            className="col-xs-12"
          >
            <div>
              
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer>
    <div
      className="container"
    >
      <div
        className="meta clearfix row"
      >
        <div
          className="col-md-12"
        >
          <ul
            className="list-inline"
          >
            <li>
              <a
                href="#tickets"
              >
                Terms & conditions
              </a>
            </li>
            <li>
              <a
                href="#tickets"
              >
                About us
              </a>
            </li>
            <li>
              <a
                href="#tickets"
              >
                Contact us
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
